### DISCUSSION TITLE
How to get latest commit from a PR merge?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#25771] [Actions]
<p>I have a GitHub Action setup, which sends a Discord Webhook notification once a commit was performed towards a specific branch, and the GitHub Pages where built using it.</p>
<p>I currently use <code>${{ github.event.commit[0].message }}</code> to display the commit message as a description in the send webhook.<br>
The issue I face here is, that it always uses the very first made Commit and not the last one made when the PR was merged.</p>
<p>For example, when I made these commits in a PR:</p>
<pre><code class="lang-nohighlight">Change value to true
Update description of docs
Minor tweaks
</code></pre>
<p>…will the Action use the very first commit, while I want to use the last one.</p>
<p>If possible would I now like to switch to displaying the title that was used in the Pull request.<br>
All commits performed are always through commits, because I have branch-protection set up. Is there a way that I could get the title of a PR from a merge, or at the very least get the latest Commit message from the push?</p>
<p>My github action can be found here:<br>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/purrbot-site/Docs/blob/1db3bbf778822ce17baca46a873cc22c1876c1b1/.github/workflows/deploy.yml" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/purrbot-site/Docs/blob/1db3bbf778822ce17baca46a873cc22c1876c1b1/.github/workflows/deploy.yml" target="_blank" rel="nofollow noopener">purrbot-site/Docs/blob/1db3bbf778822ce17baca46a873cc22c1876c1b1/.github/workflows/deploy.yml</a></h4>
<pre><code class="lang-yml">name: Deploy Site

on:
  push:
    paths: 
    - 'docs/**'
    - '**.yml'
    branches:
    - master

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - name: Chekout code
      uses: actions/checkout@v2
    - name: Set up Python 3.7
      uses: actions/setup-python@v2.0.1
      with:
        python-version: 3.7
</code></pre>

  This file has been truncated. <a href="https://github.com/purrbot-site/Docs/blob/1db3bbf778822ce17baca46a873cc22c1876c1b1/.github/workflows/deploy.yml" target="_blank" rel="nofollow noopener">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/Andre601">@Andre601</a>,</p>
<p>To get the <strong>title</strong> of the PR, you can use the expression “github.event.pull_request.title” to get it from the <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github</strong>  context</a>, when the workflow is running on the events for pull request.<br>
For example:</p>
<pre><code class="lang-yaml">echo "The title of PR is: ${{ github.event.pull_request.title }}"
</code></pre>
<p>To get the latest commit on the PR, you can do as the steps below:</p>
<ul>
<li>Use the “<a href="https://docs.github.com/en/rest/reference/pulls#list-commits-on-a-pull-request" rel="nofollow noopener"><strong>List commits on a pull request</strong></a>” API to list all the commits on current PR, and save the <strong>response</strong> as <strong>JSON</strong>.</li>
<li>Use <a href="https://stedolan.github.io/jq/manual/" rel="nofollow noopener"><strong>jq</strong> filter</a> to get the latest commit from the commits list (the list is a <strong>JSON array</strong>). Generally, the latest commit is the last element of the array, so you need to get the <strong>index</strong> (<strong>length - 1</strong>) of this element.<br>
The following is a simple demo as reference:</li>
</ul>
<pre><code class="lang-yaml">on: pull_request

jobs:
  job1:
    name: Test
    if: ${{ github.event_name == 'pull_request' }}
    runs-on: ubuntu-latest
    steps:
      - name: List commits on the pull request
        run: |
          response=$(curl --request GET \
          --url 'https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/commits' \
          --header 'Authorization: token ${{ secrets.GITHUB_TOKEN }}' \
          --header 'Accept: application/vnd.github.v3+json' \
          --header 'Content-Type: application/json')
          
          echo $response &gt; response.json

      - name: Get latest commit
        run: |
          echo "Get length of the array:"
          length=$(jq 'length' response.json)

          echo "Get index of the latest commit:"
          index=$(($length - 1))

          echo "Get context of the latest commit:"
          latest_commit=$(jq --argjson index $index '.[$index]' response.json)
          echo "$latest_commit"

          echo "Get commit message of the latest commit:"
          latest_commit_message=$(echo "$latest_commit" | jq -r '.commit.message')
          echo "$latest_commit_message"

          echo "Get commit SHA of the latest commit:"
          latest_commit_sha=$(echo "$latest_commit" | jq -r '.sha')
          echo "$latest_commit_sha"
</code></pre>

**Comment by tony** [comment]
<p>Will this actually work when a Merge is performed?   When inside an unmerged PR, this works, but when a PR is merged via Merge commit, this happens:</p>
<pre><code class="lang-auto">Run response=$(curl --request GET \
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    84  100    84    0     0   1024      0 --:--:-- --:--:-- --:--:--  1024
Run echo "Get length of the array:"
  echo "Get length of the array:"
  length=$(jq 'length' response.json)
  
  echo "Get index of the latest commit:"
  index=$(($length - 1))
  
  echo "Get context of the latest commit:"
  latest_commit=$(jq --argjson index $index '.[$index]' response.json)
  echo "$latest_commit"
  
  echo "Get commit message of the latest commit:"
  latest_commit_message=$(echo "$latest_commit" | jq -r '.commit.message')
  echo "$latest_commit_message"
  
  echo "Get commit SHA of the latest commit:"
  latest_commit_sha=$(echo "$latest_commit" | jq -r '.sha')
  echo "$latest_commit_sha"
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    SHOULD_TEST: true
Get length of the array:
Get index of the latest commit:
Get context of the latest commit:
jq: error (at response.json:1): Cannot index object with number
</code></pre>
<p>I’m surprised by how challenging this is.</p>
<p>Since my intention is slightly different than the original poster, I will also make a new topic</p>
